cli/src/description_util.rs

```diff
@@ -40,6 +45,110 @@ JJ: Lines starting with "JJ: " (like this one) will be removed.
     Ok(text_util::complete_newline(description.trim_matches('\n')))
 }
 
+#[derive(Debug)]
+pub struct EditMultipleDescriptionsResult {
+    /// The parsed, formatted descriptions.
+    pub descriptions: HashMap<CommitId, String>,
+    /// Commit IDs that were found multiple times while parsing the edited
+    /// messages.
+    pub duplicates: Vec<String>,
+    /// Commit IDs that were found while parsing the edited messages, but which
+    /// were not originally being edited.
+    pub unexpected: Vec<String>,
+}
+
+/// Edits the descriptions of the given commits in a single editor session.
+pub fn edit_multiple_descriptions(
+    ui: &Ui,
+    settings: &UserSettings,
+    workspace_command: &WorkspaceCommandHelper,
+    repo: &ReadonlyRepo,
+    commits: &[Commit],
+) -> Result<EditMultipleDescriptionsResult, CommandError> {
+    let id_prefix_context = workspace_command.id_prefix_context()?;
+    let mut commits_map = HashMap::new();
+    let mut output_chunks = Vec::new();
+
+    for commit in commits.iter().rev() {
+        let commit_id_prefix = generate_commit_id_prefix(id_prefix_context, repo, commit);
+        if commits.len() > 1 {
+            output_chunks.push(format!("JJ: describe {}\n", commit_id_prefix.clone()));
+        }
+        commits_map.insert(commit_id_prefix, commit);
+        let template = description_template_for_describe(ui, settings, workspace_command, commit)?;
+        output_chunks.push(template);
+        output_chunks.push("\n".to_owned());
+    }
+    output_chunks
+        .push("JJ: Lines starting with \"JJ: \" (like this one) will be removed.\n".to_owned());
+    let bulk_message = output_chunks.join("");
+
+    let bulk_message = edit_temp_file(
+        "description",
+        ".jjdescription",
+        repo.repo_path(),
+        &bulk_message,
+        settings,
+    )?;
+
+    let bulk_message = match commits {
+        // For single commits, add the marker line, but only if the user hasn't already done so.
+        [commit] if !bulk_message.contains("JJ: describe") => {
+            let commit_id_prefix = generate_commit_id_prefix(id_prefix_context, repo, commit);
+            format!("JJ: describe {}\n{}", commit_id_prefix, bulk_message)
+        }
+        _ => bulk_message,
+    };
+
+    // Split the bulk message into (commit_id_prefix, message) tuples
+    let msgs = bulk_message
+        .split("JJ: describe")
```

Comment:
nit: "\nJJ: describe"? (or process by line)

It might be also good to extract parsing function and add test for it.

Reply:
That makes sense, I just wasn't sure what to use as the delimiter. `git reword`'s `++` seems good to me. Any other opinions?

Reply:
I don't have a better idea at present. I think we can land it as-is and follow up.

There might be some kind of useful syntax that editors could use to denote regions. Some Vim and Emacs users might have something to say about it? (Can those editors denote regions without matched opening/closing pairs? Is there any particular syntax structure we should try to use so that those editors can easily configure/detect such regions?)

It might actually be best to keep the comments for now so that the maintainers of editor integrations don't have to adjust the parsing rules again if we were to change the delimiters later.